[Tooling] Update `release-toolkit` to fix milestone moving issue

iangmaia opened the pull request:
Follow-up to AINFRA-1359

### Description

This PR uses the [13.5.1](https://github.com/wordpress-mobile/release-toolkit/releases/tag/13.5.1) release-toolkit version that fixes the action `update_assigned_milestone` to work with fine-grained GitHub tokens (see PR https://github.com/wordpress-mobile/release-toolkit/pull/663).

iangmaia committed:
Update release-toolkit to fix milestone moving issue

dangermattic commented:
<!--
  1 Error: PR is not assigned to a milest...
  0 Warnings
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

AliSoftware commented on Gemfile:
```diff
@@ -9,7 +9,9 @@ gem 'rubocop', '~> 1.65'
 ### Fastlane Plugins
 
-gem 'fastlane-plugin-wpmreleasetoolkit', '~> 12.3'
+# The '>= 13.5.1' after '~> 13.5' ensures that we resolve to any version compatible with 13.5 starting from 13.5.1
+# We need 13.5.1 because of a fix in fetching issues / PRs with a fine-grained GitHub token
+gem 'fastlane-plugin-wpmreleasetoolkit', '~> 13.5', '>= 13.5.1'
```
💄  This is fine, though tbh `~> 13.5` could also have been enough as long as you made sure to re-run `bundle update fastlane-plugin-wpmreleasetoolkit` once `13.5.1` was deployed so that the dependency resolution would discover it and pick it.
(As long as the `Gemfile.lock` lockfile is on a version that is `>=13.5.1` after you do such update, there's no way it would risk regressing to `13.5.0` later on)

But I don't mind being explicit in the `Gemfile` even if the `Gemfile.lock` would have already resolved it to `13.5.1`, both are valid anyway 👍

iangmaia replied:
Hmm yeah I was basically following what the `Gemfile` on iOS [had](https://github.com/woocommerce/woocommerce-ios/pull/16221#discussion_r2412204648), I've seen this pattern in a few of our Gemfiles. Tbh I'd prefer not to be redundant... and these comments are bound to be out of date / not relevant anyway while dependencies are getting updated (like the previous iOS one did).

iangmaia replied:
Updated on https://github.com/woocommerce/woocommerce-android/pull/14715/commits/acf8e1bed4b1392af690202996d98b485e9e8ca3, favoring a clean-up.

iangmaia committed:
Remove `>= 13.5.1` requirement and clean Gemfile and run `bundle update fastlane-plugin-wpmreleasetoolkit`

h/t @AliSoftware https://github.com/woocommerce/woocommerce-android/pull/14715#discussion_r2412106440
